### ISSUE TITLE
"*.views.dll" need to recompile

### ISSUE THREAD

**Issue by cqkisyouq** [aspnet/AspNetCore#11836]
### Describe the bug
"*.views.dll" Added to the program as a module, Program works. When I made changes to the view in the program while I was developing, I was told that I couldn't compile the module.

"TestModule1"  It's my view module:

      Connection id "0HLNVHL8KMNU3", Request id "0HLNVHL8KMNU3:00000001": An unhandled exception was thrown by the application.
System.InvalidOperationException: Cannot find compilation library location for package 'TestModule1'
   at Microsoft.Extensions.DependencyModel.CompilationLibrary.ResolveReferencePaths(ICompilationAssemblyResolver resolver, List`1 assemblies)
   at Microsoft.Extensions.DependencyModel.CompilationLibrary.ResolveReferencePaths()
   at Microsoft.AspNetCore.Mvc.ApplicationParts.AssemblyPart.<>c.<GetReferencePaths>b__8_0(CompilationLibrary library)
   at System.Linq.Enumerable.SelectManySingleSelectorIterator`2.MoveNext()
   at Microsoft.AspNetCore.Mvc.Razor.Compilation.MetadataReferenceFeatureProvider.PopulateFeature(IEnumerable`1 parts, MetadataReferenceFeature feature)
   at Microsoft.AspNetCore.Mvc.ApplicationParts.ApplicationPartManager.PopulateFeature[TFeature](TFeature feature)
   at Microsoft.AspNetCore.Mvc.Razor.Internal.DefaultRazorReferenceManager.GetCompilationReferences()
   at System.Threading.LazyInitializer.EnsureInitializedCore[T](T& target, Boolean& initialized, Object& syncLock, Func`1 valueFactory)
   at Microsoft.AspNetCore.Mvc.Razor.Internal.DefaultRazorReferenceManager.get_CompilationReferences()
   at Microsoft.AspNetCore.Mvc.Razor.Internal.LazyMetadataReferenceFeature.get_References()
   at Microsoft.CodeAnalysis.Razor.CompilationTagHelperFeature.GetDescriptors()
   at Microsoft.AspNetCore.Razor.Language.DefaultRazorTagHelperBinderPhase.ExecuteCore(RazorCodeDocument codeDocument)
   at Microsoft.AspNetCore.Razor.Language.RazorEnginePhaseBase.Execute(RazorCodeDocument codeDocument)
   at Microsoft.AspNetCore.Razor.Language.DefaultRazorEngine.Process(RazorCodeDocument document)
   at Microsoft.AspNetCore.Razor.Language.DefaultRazorProjectEngine.ProcessCore(RazorCodeDocument codeDocument)
   at Microsoft.AspNetCore.Razor.Language.RazorProjectEngine.Process(RazorProjectItem projectItem)
   at Microsoft.AspNetCore.Mvc.Razor.Internal.RazorViewCompiler.CompileAndEmit(String relativePath)
   at Microsoft.AspNetCore.Mvc.Razor.Internal.RazorViewCompiler.OnCacheMiss(String normalizedPath)
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.AspNetCore.Mvc.RazorPages.Internal.DefaultPageLoader.Load(PageActionDescriptor actionDescriptor)
   at Microsoft.AspNetCore.Mvc.RazorPages.Internal.PageActionInvokerProvider.OnProvidersExecuting(ActionInvokerProviderContext context)
   at Microsoft.AspNetCore.Mvc.Internal.ActionInvokerFactory.CreateInvoker(ActionContext actionContext)
   at Microsoft.AspNetCore.Mvc.Internal.MvcAttributeRouteHandler.<>c__DisplayClass12_0.<RouteAsync>b__0(HttpContext c)
   at Microsoft.AspNetCore.Builder.RouterMiddleware.Invoke(HttpContext httpContext)
   at Microsoft.AspNetCore.StaticFiles.StaticFileMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.HttpProtocol.ProcessRequests[TContext](IHttpApplication`1 application)

**Comment by cqkisyouq** [comment]
Add modules to your program:

foreach (var item in moduleManager.GetModules())
            {
                if (item.Assembly != null) mvcBuilder.AddApplicationPart(item.Assembly);
                if (item.ViewAssembly != null)
                {
                    foreach (var view in CompiledRazorAssemblyApplicationPartFactory.GetDefaultApplicationParts(item.ViewAssembly))
                    {
                        mvcBuilder.ConfigureApplicationPartManager(apm => apm.ApplicationParts.Add(view));
                    }
                }
            }

**Comment by mkArtakMSFT** [comment]
Hi. It looks like this is a question about how to use ASP.NET Core. While we do our best to look through all the issues filed here, to get a faster response we suggest posting your questions to [StackOverflow](https://www.stackoverflow.com) using the **asp.net-core-mvc** tag.

**Comment by cqkisyouq** [comment]
I load "*.views. DLL "into the current program in a dynamic way. When I run the debugging of the current program and do not change the views in the current program, everything is normal when visiting the page.
When I make a change to "*.cshtml" in the current program and save it, and then access the changed view without restarting the debugging, I will report the above error.
By looking at the above error analysis, when I changed save "*.cshtml", the view was updated and recompiled.
But "*.views. DLL "is the view file in another library I dynamically reference. At this time, when recompiling the view, I also compile the dynamically referenced view "*.views.
They're not the same project anyway, so I don't think they should compile the view module I'm referring to.
